src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -112,26 +116,39 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             SyntaxGenerator generator = SyntaxGenerator.GetGenerator(doc);
 
-            // The stream object
+            // The stream-derived instance
             SyntaxNode streamInstanceNode = invocation.Instance.Syntax;
 
-            // buffer.AsMemory(int start, int length)
-            // offset should become start
-            // count should become length
-            SyntaxNode namedStartNode = isOffsetNamed ? generator.Argument(name: "start", RefKind.None, offsetValueNode) : offsetValueNode;
-            SyntaxNode namedLengthNode = isCountNamed ? generator.Argument(name: "length", RefKind.None, countValueNode) : countValueNode;
-
-            // Generate an invocation of the AsMemory() method from the byte array object, using the correct named arguments
-            SyntaxNode asMemoryExpressionNode = generator.MemberAccessExpression(bufferValueNode.WithoutTrivia(), memberName: "AsMemory");
-            SyntaxNode asMemoryInvocationNode = generator.InvocationExpression(
-                asMemoryExpressionNode,
-                namedStartNode.WithTriviaFrom(offsetValueNode),
-                namedLengthNode.WithTriviaFrom(countValueNode));
+            // Depending on the arguments being passed to Read/WriteAsync, it's the substitution we will make
+            SyntaxNode replacedInvocationNode;
 
-            // Generate the new buffer argument, ensuring we include the argument name if the user originally indicated one
-            SyntaxNode namedBufferWithAsMemoryInvocationNode =
-                    (isBufferNamed ? generator.Argument(name: "buffer", RefKind.None, asMemoryInvocationNode) : asMemoryInvocationNode)
-                .WithTriviaFrom(bufferValueNode);
+            if(IsPassingZeroAndBufferLength(bufferValueNode, offsetValueNode, countValueNode))
+            {
+                // Remove 0 and buffer.length
+                replacedInvocationNode =
+                        (isBufferNamed ? generator.Argument(name: "buffer", RefKind.None, bufferValueNode) : bufferValueNode)
```

Comment:
I have noticed you are using a lot the following pattern: `boolVar ? generator.Argument(name: someName, RefKind.None, node) : node`. You could probably extract it as a function taking the 3 parameters.

Reply:
I'm using the conditional operator _precisely_ because the checks are very simple and don't deserve their own function.